tests/infrastructure/provisioners/test_container_instance.py

```diff
@@ -0,0 +1,654 @@
+from subprocess import CalledProcessError
+from unittest.mock import AsyncMock, MagicMock, call
+
+import pytest
+
+from prefect.infrastructure.provisioners.container_instance import (
+    ContainerInstancePushProvisioner,
+)
+
+
```

Comment:
Can you add tests for these cases:
1. Call `.provision`, `az` not installed
2. Call `.provision`, no preexisting credentials block
3. Call `.provision`, existing credentials block

In the last two cases, we should check to ensure the base job template is configured with the expected resources.

Reply:
Will do!

Reply:
I've applied those changed in subsequent commits :)